src/MediatR/Mediator.cs

```diff
@@ -17,6 +17,9 @@ public class Mediator : IMediator
 
         private readonly MultiInstanceFactory _multiInstanceFactory;
 
+        private readonly Dictionary<Type, Type> genericHandlerCache;
+        private readonly Dictionary<Type, Type> wrapperHandlerCache;

```

Comment:
@khellang Just historical. I'm moving to .NET platform standard eventually, the question is whether to make a pitstop at `dotnet5.1` or not (I've had problems with it). It probably wouldn't hurt just to jump to it.